projects/components/src/action-menu/action-menu.component.ts

```diff
@@ -77,6 +68,12 @@ export class ActionMenuComponent<R, T> {
         return this._actionDisplayConfig;
     }
 
+    /**
+     * Copy of actions passed with their availability call backs. This is because, when the selected entities get updated, we need to use
+     * those CBs to calculate the availability again
+     */
+    private actionsWithAvailabilityCb: ActionItem<R, T>[];
```

Comment:
Its confusing that some filtering happens in the template and some happens in the component. There are also a few methods that exist that I'm not sure need to.

I think this would be more clear if the following was true:
1. Action item availability is input as either a boolean, a function, or an observable
2. We convert availability to always be an Observable and store it in this private variable (`actionsWithAvailabilityAsObs`)
3. This removes all calls to `isObservable` in the templates, and cleans up methods like `isActionAvailable`.
4. In dropdown.component.ts, we now require that all availability is input as an observable. This again cleans up the code. 

So pretty much, accepting availability as a boolean or a function is a convenience thing for clients, and we convert everything to an observable. This will clean up the template code, and the typescript code too.

We might want to delay this so we can get the feature in by beta though... if so, let's create a follow up ticket and do it right away

Reply:
> We convert everything to an observable

As discussed with Ryan on zoom call, I think this is going to be problematic for the following reasons:
- We have to create new observables everytime the callbacks/booleans get updated because of changes in selected entities or actions list
- The calculation of availability is required on the TS side for UX things like:
1. Ungrouping single nested actions
2. The number of featured actions shown should be equal to the featured count. For example, In the following scenario where Featured actions: 3, Featured count: 2, If the 2nd featured action is unavailable, and we don't filter that, we only show the first action in featured bar!
- So, the code calling available actions in the class should become async and this makes the collection(action list) processing complicated. And I don't see this as necessary right now.

If we still want to explore this approach, I created a task to do it later: https://jira.eng.vmware.com/browse/VDUCC-615